libr/anal/anal.c

```diff
@@ -551,6 +552,7 @@ R_API bool r_anal_noreturn_add(RAnal *anal, const char *name, ut64 addr) {
 		}
 		tmp_name = fcn ? fcn->name: fi->name;
 		if (fcn) {
+			anal->is_dirty = fcn->is_noreturn? true : anal->is_dirty;
```

Comment:
Use an if, otherwise when this variable is true is then set to true again in the line below

Reply:
I would have thought that the compiler optimizes that way